Projects/Delivery/ApiHandlers/src/com/ilmlf/delivery/api/handlers/util/SecretsUtil.java

```diff
@@ -0,0 +1,36 @@
+package com.ilmlf.delivery.api.handlers.util;
+
+import org.json.JSONObject;
+import lombok.NonNull;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+import software.amazon.awssdk.services.secretsmanager.model.SecretsManagerException;
+
+public abstract class SecretsUtil {
+
+  public static JSONObject getSecret(@NonNull String region, @NonNull String dbSecretName) {
+    SecretsManagerClient secretsClient = SecretsManagerClient.builder()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *secretsClient*. The resource is referenced at line 26. The resource is closed at line 29. There are other execution paths that do not contain closure statements, for example, when GetSecretValueResponse.secretString throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).